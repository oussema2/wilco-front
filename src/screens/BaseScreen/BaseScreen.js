import React from 'react';
import PropTypes from 'prop-types';
import { observer } from 'mobx-react-lite';
import { StatusBar } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { ScreenGradient } from '../../components/ScreenGradient';
import { ScreenLoader } from '../../components/ScreenLoader';
import { styles } from './BaseScreen.styles';
import useKeyboardDismiss from '../../hooks/useKeyboardDismiss';

const BaseScreen = ( {
	children, testID, isLoading, edgeTop
} ) => {
	useKeyboardDismiss( isLoading );

	const _defaultEdges = [ 'top', 'right', 'bottom', 'left' ];
	return (
		<ScreenGradient style={styles.containerView} testID={testID}>
			<StatusBar translucent />
			<SafeAreaView
				edges={( edgeTop ) ? [ 'top' ] : _defaultEdges}
				style={styles.safeArea}
			>
				{children}
			</SafeAreaView>
			{isLoading && <ScreenLoader />}
		</ScreenGradient>
	);
};

BaseScreen.propTypes = {
	testID: PropTypes.string,
	children: PropTypes.oneOfType( [
		PropTypes.bool,
		PropTypes.element,
		PropTypes.node
	] ),
	isLoading: PropTypes.bool,
	edgeTop: PropTypes.bool
};

BaseScreen.defaultProps = {
	testID: 'base-screen',
	children: <></>,
	isLoading: false,
	edgeTop: false
};

export default observer( BaseScreen );
