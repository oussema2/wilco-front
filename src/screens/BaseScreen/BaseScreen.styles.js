import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create( {
	containerView: {
		width: '100%',
		height: '100%',
		display: 'flex',
		flexDirection: 'column',
		justifyContent: 'center',
		alignItems: 'center'
	},
	safeArea: {
		width: '100%',
		height: '100%'
	}
} );
