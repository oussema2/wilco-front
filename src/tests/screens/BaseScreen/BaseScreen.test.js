import * as React from 'react';
import { render } from '@testing-library/react-native';
import { View } from 'react-native';
import BaseScreen from '../../../screens/BaseScreen/BaseScreen';

describe( 'BaseScreen', () => {
	let screen;
	const testID = 'testID';
	const children = <View testID="view" />;

	const setUp = ( ) => {
		screen = render( <BaseScreen testID={testID}>{children}</BaseScreen> );
	};

	beforeEach( () => {
		setUp();
	} );

	describe( 'isLoading', () => {
		describe( 'without the isLoading prop', () => {
			it( 'renders the BaseScreen screen correctly', async () => {
				expect( screen.queryByTestId( testID ) ).not.toBeNull();
				expect( screen.queryByTestId( 'view' ) ).not.toBeNull();
				expect( screen.queryByTestId( 'screenLoader' ) ).toBeNull();

				expect( screen ).toMatchSnapshot();
			} );
		} );

		describe( 'with the isLoading prop', () => {
			beforeEach( () => {
				screen = render( <BaseScreen testID={testID} isLoading>{children}</BaseScreen> );
			} );

			it( 'renders the BaseScreen screen with a screen loader', async () => {
				expect( screen.queryByTestId( 'screenLoader' ) ).not.toBeNull();
				expect( screen ).toMatchSnapshot();
			} );
		} );
	} );

	describe( 'edgeTop', () => {
		describe( 'when edgeTop is true', () => {
			beforeEach( () => {
				screen = render( <BaseScreen edgeTop>{children}</BaseScreen> );
			} );
			it( 'renders the BaseScreen screen correctly', async () => {
				expect( screen ).toMatchSnapshot();
			} );
		} );

		describe( 'with the edgeTop prop', () => {
			it( 'renders the BaseScreen screen correctly', async () => {
				expect( screen ).toMatchSnapshot();
			} );
		} );
	} );
} );
