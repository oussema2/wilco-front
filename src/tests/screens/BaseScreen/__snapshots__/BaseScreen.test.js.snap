// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseScreen edgeTop when edgeTop is true renders the BaseScreen screen correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4291295231,
      4294967295,
    ]
  }
  endPoint={
    Object {
      "x": 0.25,
      "y": 0.3,
    }
  }
  locations={
    Array [
      0.5,
      1,
    ]
  }
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 5,
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
  testID="base-screen"
>
  <RNCSafeAreaView
    edges={
      Array [
        "top",
      ]
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      testID="view"
    />
  </RNCSafeAreaView>
</BVLinearGradient>
`;

exports[`BaseScreen edgeTop with the edgeTop prop renders the BaseScreen screen correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4291295231,
      4294967295,
    ]
  }
  endPoint={
    Object {
      "x": 0.25,
      "y": 0.3,
    }
  }
  locations={
    Array [
      0.5,
      1,
    ]
  }
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 5,
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
  testID="testID"
>
  <RNCSafeAreaView
    edges={
      Array [
        "top",
        "right",
        "bottom",
        "left",
      ]
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      testID="view"
    />
  </RNCSafeAreaView>
</BVLinearGradient>
`;

exports[`BaseScreen isLoading with the isLoading prop renders the BaseScreen screen with a screen loader 1`] = `
<BVLinearGradient
  colors={
    Array [
      4291295231,
      4294967295,
    ]
  }
  endPoint={
    Object {
      "x": 0.25,
      "y": 0.3,
    }
  }
  locations={
    Array [
      0.5,
      1,
    ]
  }
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 5,
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
  testID="testID"
>
  <RNCSafeAreaView
    edges={
      Array [
        "top",
        "right",
        "bottom",
        "left",
      ]
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      testID="view"
    />
  </RNCSafeAreaView>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.6)",
        "display": "flex",
        "flexDirection": "column-reverse",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
    testID="screenLoader"
  >
    <View
      animated={true}
      animationConfig={
        Object {
          "bounciness": 0,
        }
      }
      animationType="spring"
      indeterminate={true}
      indeterminateAnimationDuration={1000}
      onLayout={[Function]}
      progress={0}
      style={
        Array [
          Object {
            "backgroundColor": undefined,
            "borderColor": "#12A3BE",
            "borderRadius": 0,
            "borderWidth": 0,
            "overflow": "hidden",
            "width": null,
          },
          undefined,
        ]
      }
      testID="progress-bar"
      useNativeDriver={false}
    >
      <View
        style={
          Object {
            "backgroundColor": "#12A3BE",
            "height": 6,
            "transform": Array [
              Object {
                "translateX": -0,
              },
              Object {
                "translateX": -0,
              },
              Object {
                "scaleX": 0.30007,
              },
            ],
          }
        }
      />
    </View>
  </View>
</BVLinearGradient>
`;

exports[`BaseScreen isLoading without the isLoading prop renders the BaseScreen screen correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4291295231,
      4294967295,
    ]
  }
  endPoint={
    Object {
      "x": 0.25,
      "y": 0.3,
    }
  }
  locations={
    Array [
      0.5,
      1,
    ]
  }
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 5,
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
  testID="testID"
>
  <RNCSafeAreaView
    edges={
      Array [
        "top",
        "right",
        "bottom",
        "left",
      ]
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      testID="view"
    />
  </RNCSafeAreaView>
</BVLinearGradient>
`;
